hackathon_site/dashboard/frontend/src/slices/hardware/cartSlice.ts

```diff
@@ -26,6 +30,47 @@ export const cartReducerName = "cart";
 export const initialState = cartAdapter.getInitialState(extraState);
 export type CartState = typeof initialState;
 
+// Thunk
+interface RejectValue {
+    status: number;
+    message: any;
+}
+
+export const submitOrder = createAsyncThunk<
+    APIListResponse<Hardware>,
```

Comment:
This endpoint returns a custom response format, not a hardware list. Not sure if it's documented properly in swagger, but you can make a request and look at the response. It will contain what succeeded, as well as the list of errors and warnings.

[This](https://github.com/ieeeuoft/hackathon-template/blob/develop/hackathon_site/hardware/serializers.py#L260) is the serializer for the successful response, which contains errors that didn't stop the request. You can see how it gets built starting [here](https://github.com/ieeeuoft/hackathon-template/blob/develop/hackathon_site/hardware/serializers.py#L219).